src/service/repo/anonymousMessageRepo.ts

```diff
@@ -150,11 +154,16 @@ async function setChannelAsRead(channelId: string): Promise<boolean> {
   }
 }
 
-async function getAnonymousChannelDetail(channelType: GetstreamChannelType, channelId: string) {
+async function getAnonymousChannelDetail(
+  channelType: GetstreamChannelType,
+  channelId: string,
+  timeStamp?: string
+) {
+  const url = timeStamp
+    ? `${baseUrl.getAnonymousChannelDetail(channelType, channelId)}?last_fetch_date=${timeStamp}`
+    : baseUrl.getAnonymousChannelDetail(channelType, channelId);
   try {
-    const response = await anonymousApi.get(
-      baseUrl.getAnonymousChannelDetail(channelType, channelId)
-    );
+    const response = await anonymousApi.get(url);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here introduce a new optional parameter `timeStamp` to the `getAnonymousChannelDetail` function. This is used to conditionally construct the URL for fetching channel details. However, there's no error handling in case the `timeStamp` parameter is not a valid date string. It would be good to add some validation and error handling for this.

```diff
async function getAnonymousChannelDetail(
  channelType: GetstreamChannelType,
  channelId: string,
  timeStamp?: string
) {
+    if (timeStamp && isNaN(Date.parse(timeStamp))) {
+      return Promise.reject(new Error('Invalid timestamp'));
+    }
  const url = timeStamp
    ? `${baseUrl.getAnonymousChannelDetail(channelType, channelId)}?last_fetch_date=${timeStamp}`
    : baseUrl.getAnonymousChannelDetail(channelType, channelId);
   try {
    const response = await anonymousApi.get(url);
```
This will ensure that an invalid `timeStamp` does not lead to unexpected behavior or errors when making the API call.


<!-- This is an auto-generated comment by OSS CodeRabbit -->